abci/types/result.go

```diff
@@ -43,14 +43,34 @@ func (r ResponseQuery) IsErr() bool {
 	return r.Code != CodeTypeOK
 }
 
+// IsUnknown returns true if Code is Unknown
+func (r ResponseVerifyVoteExtension) IsUnknown() bool {
+	return r.Result == ResponseVerifyVoteExtension_UNKNOWN
+}
+
 // IsOK returns true if Code is OK
 func (r ResponseVerifyVoteExtension) IsOK() bool {
-	return r.Result <= ResponseVerifyVoteExtension_ACCEPT
+	return r.Result == ResponseVerifyVoteExtension_ACCEPT
 }
 
 // IsErr returns true if Code is something other than OK.
 func (r ResponseVerifyVoteExtension) IsErr() bool {
-	return r.Result > ResponseVerifyVoteExtension_ACCEPT
+	return r.Result != ResponseVerifyVoteExtension_ACCEPT
+}
+
+// IsUnknown returns true if Code is Unknown
+func (r ResponseProcessProposal) IsUnknown() bool {
+	return r.Result == ResponseProcessProposal_UNKNOWN
+}
+
+// IsOK returns true if Code is OK
+func (r ResponseProcessProposal) IsOK() bool {
+	return r.Result == ResponseProcessProposal_ACCEPT
+}
+
+// IsErr returns true if Code is something other than OK.
+func (r ResponseProcessProposal) IsErr() bool {
+	return r.Result != ResponseProcessProposal_ACCEPT
 }
```

Comment:
It seems to be a bit redundant to have these different response types when we only check for `IsOk()`. Either remove these or we at least log the different result types

Reply:
Totally agree.